src/components/tl/cuda/alltoallv/alltoallv_ce.c

```diff
@@ -0,0 +1,405 @@
+            !ucc_tl_cuda_team_topo_is_direct(&team->super, team->topo,
+                                             UCC_TL_TEAM_RANK(team), i)) {
+            continue;
+        }
+        peer_sync = TASK_SYNC(task, i);
+        cache     = ucc_tl_cuda_get_cache(team, i);
+        if (ucc_unlikely(!cache)) {
+            status = UCC_ERR_NO_MESSAGE;
+            goto exit_err;
+        }
+        status = ucc_tl_cuda_map_memhandle(
+            peer_sync->mem_info_src.ptr, peer_sync->mem_info_src.length,
+            peer_sync->mem_info_src.handle,
+            &task->alltoallv_ce.peer_map_addr_src[i], cache);
+        if (UCC_OK != status) {
+            ucc_error("ucc_cuda_ipc_map_memhandle failed");
+            return UCC_ERR_INVALID_PARAM;
+        }
+        status = ucc_tl_cuda_map_memhandle(
+            peer_sync->mem_info_dst.ptr, peer_sync->mem_info_dst.length,
+            peer_sync->mem_info_dst.handle,
+            &task->alltoallv_ce.peer_map_addr_dst[i], cache);
+        if (UCC_OK != status) {
+            ucc_error("ucc_cuda_ipc_map_memhandle failed");
+            return UCC_ERR_INVALID_PARAM;
+        }
+    }
+    return UCC_OK;
+
+exit_err:
+    return status;
+}
+
+ucc_status_t ucc_tl_cuda_alltoallv_ce_post_copies(ucc_tl_cuda_task_t *task)
+{
+    ucc_tl_cuda_team_t         *team = TASK_TEAM(task);
+    ucc_rank_t                  rank = UCC_TL_TEAM_RANK(team);
+    ucc_tl_cuda_sync_t         *sync = TASK_SYNC(task, rank);
+    ucc_tl_cuda_sync_t         *peer_sync;
+    ucc_ee_executor_t          *exec;
+    void                       *src, *dst;
+    ucc_ee_executor_task_t    **exec_task;
+    size_t                      data_size, data_displ;
+    ucc_rank_t                  i, peer, psrc, pdst;
+    ucc_status_t                status;
+    ucc_ee_executor_task_args_t exec_args;
+
+    task->alltoallv_ce.num_posted = 0;
+    status = ucc_coll_task_get_executor(&task->super, &exec);
+    if (ucc_unlikely(status != UCC_OK)) {
+        goto exit;
+    }
+
+    for (i = 0; i < UCC_TL_TEAM_SIZE(team); i++) {
+        peer = (rank + i) % UCC_TL_TEAM_SIZE(team);
+        if (!ucc_tl_cuda_team_topo_is_direct(&team->super, team->topo, rank,
+                                             peer)) {
+            continue;
+        }
+        peer_sync = TASK_SYNC(task, peer);
+        if (peer == rank) {
+            src = task->alltoallv_ce.sbuf;
+        } else {
+            src = PTR_OFFSET(task->alltoallv_ce.peer_map_addr_src[peer],
+                             peer_sync->mem_info_src.offset);
+            CUDA_CHECK_GOTO(
+                cudaStreamWaitEvent(team->stream,
+                                    sync->data[peer].ipc_event_remote, 0),
+                exit, status);
+        }
+        data_size =
+            task->alltoallv_ce.get_size(task, sync->alltoallv_ce.sbytes, peer);
+        data_displ = task->alltoallv_ce.get_offset(
+            task, sync->alltoallv_ce.sdispl_bytes, peer);
```

Comment:
data_displ is not correct
```suggestion
        data_size =
            task->alltoallv_ce.get_size(task, peer_sync->alltoallv_ce.sbytes, rank);
        data_displ = task->alltoallv_ce.get_offset(
            task, peer_sync->alltoallv_ce.sdispl_bytes, rank);
```